fix(android): app/module builds fail with JDK 13

jquick-axway opened the pull request:
**JIRA:**
- https://jira.appcelerator.org/browse/TIMOB-27778
- https://jira.appcelerator.org/browse/TIMOB-27780
- https://jira.appcelerator.org/browse/TIMOB-27781

**Summary:**
- [TIMOB-27778](https://jira.appcelerator.org/browse/TIMOB-27778) Updated gradle build tools from 3.4.2 to 3.6.0.
- [TIMOB-27780](https://jira.appcelerator.org/browse/TIMOB-27780) Hyperloop builds fail if JDK 12 or higher is installed.
  * Updated hyperloop module version from 5.0.1 to 5.0.2.
- [TIMOB-27781](https://jira.appcelerator.org/browse/TIMOB-27781) App/Module builds will fail with Titanium 9.0.0 and JDK 13.
  * This is a regression. Titanium 8.x.x used to build fine with JDK 13 (except hyperloop).
  * Fixed by updating gradle wrapper/library from 5.4.1 to 6.2.1.
  * Gradle added JDK 13 support in version 6.0.0.

---
**New Build Warning:**
App builds will now log the following deprecation warning. This is because gradle 6.0.0 has deprecated an API that Google's gradle build tool is currently using. We're stuck with this until Google resolves it their 4.0.0 build tool which hasn't been released yet.
```
[INFO]  [GRADLE] Internal API constructor DefaultDomainObjectSet(Class<T>) has been deprecated. This is scheduled to be removed in Gradle 7.0. Please use ObjectFactory.domainObjectSet(Class<T>) instead. See https://docs.gradle.org/6.2.1/userguide/custom_gradle_types.html#domainobjectset for more details.
```

---
**Note to QE:**
On Mac, you can have multiple JDKs installed. After installing JDK 13 and running the below tests, you can select the older installed JDK by doing the following at the command line.

List the JDK versions installed on Mac.
```
/usr/libexec/java_home -V
```

Select the JDK version to use. You should set this in your `.bash_profile` file. Replace below `<JavaVersion>` with what was listed above.
```bash
export JAVA_HOME=$(/usr/libexec/java_home -v <JavaVersion>)
```

---
**App Build Test:**
1. Set up a machine with JDK 13. (Reboot if you just installed it.)
2. Build [kitchensink-v2](https://github.com/appcelerator/kitchensink-v2) for Android.
3. Verify it builds and runs on Android without issue.

---
**Module Build Test:**
1. Set up a machine with JDK 13. (Reboot if you just installed it.)
2. Download a zip of the [ti.imagefactory](https://github.com/appcelerator-modules/ti.imagefactory) source.
3. Open the Terminal.
4. `CD` to directory: `./ti.imagefactory/android`
5. Enter: `appc run -p android`
6. Verify module builds and its example project runs in the emulator.

---
**Hyperloop Build Test:**
1. Set up a machine with JDK 12 or higher. (Reboot if you just installed it.)
2. Download the [hyperloop-example](https://github.com/appcelerator/hyperloop-examples) project.
3. Build for Android.
4. Verify app successfully builds and runs on device.

jquick-axway committed:
chore(android): update gradle build tools to 3.6.0

- Added "android.useAndroidX" gradle property which is now requried or else a build failure will occur.
- Resolved Java and C/C++ build order issue with newest tool.

jquick-axway committed:
fix(android): updated gradle wrapper so it can build with JDK 13

- Updated gradle wrapper/library to version 6.2.1. (Gradle added JDK 13 support in 6.0.0.)

build commented:
<!--
  0 failure: 
  0 warning: 
  3 messages
  
  DangerID: danger-id-Danger;
-->



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-11507/1/artifact/dist/mobilesdk-9.1.0.v20200228202816-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:fist: The commits in this PR match our conventions! Feel free to Rebase and Merge this PR when ready.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :white_check_mark: All tests are passing
Nice one! All 6555 tests are passing.
(There are 703 skipped tests not included in that total)
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 330406d83d2d91b1c0c6b54c718c72ee154d40c5
</p>

jquick-axway committed:
fix(android): hyperloop builds fail with NDK 12 or higher

- JDK 12 dropped support for Java 6 language support. Hyperloop now compiles with Java 8 language.
- Updated hyperloop module version from 5.0.1 to 5.0.2.

garymathews reviewed (approved):
LGTM!

jquick-axway committed:
Merge branch 'master' into TIMOB-27778

ssaddique committed:
Merge branch 'master' into TIMOB-27778

ssaddique committed:
Merge branch 'master' into TIMOB-27778

ssaddique commented:
FR partially passed. Please see JIRA ticket for details

ssaddique committed:
Merge branch 'master' into TIMOB-27778

ssaddique committed:
Merge branch 'master' into TIMOB-27778

ssaddique committed:
Merge branch 'master' into TIMOB-27778

ssaddique committed:
Merge branch 'master' into TIMOB-27778

ssaddique commented:
**FR: Pass**

ssaddique committed:
Merge branch 'master' into TIMOB-27778
